src/cleanupCommand/cleanupManager.ts

```diff
@@ -96,14 +128,64 @@ export class CleanupManager {
     return filteredTasks;
   }
 
-  private async deleteExpiredFailedTasksSources(tasks: IJob<IngestionParams>[]): Promise<IJob<IngestionParams>[]> {
+  private async filterFromRunningExportJobs(jobs: IJob<IWithCleanDataIngestionParams>[]): Promise<IJob<IWithCleanDataIngestionParams>[]> {
+    const runningExportJobs = await this.jobManager.getInProgressJobs(this.exporterJobType);
+    const cleanupReadyJobs: IJob<IWithCleanDataIngestionParams>[] = [];
+
+    for (const swapJob of jobs) {
+      const cleanupData = swapJob.parameters.cleanupData;
+      const filteredJobs = runningExportJobs.filter(
+        (job) => cleanupData?.previousProductVersion === job.version && swapJob.parameters.metadata.productId === job.resourceId
+      );
+      if (!filteredJobs.length) {
+        cleanupReadyJobs.push(swapJob);
+      }
+    }
+    return cleanupReadyJobs;
+  }
+
+  private async deleteExpiredFailedTasksSources(tasks: IJob<IWithCleanDataIngestionParams>[]): Promise<IJob<IWithCleanDataIngestionParams>[]> {
```

Comment:
add info log for deleted expired sources

Reply:
?

Reply:
done